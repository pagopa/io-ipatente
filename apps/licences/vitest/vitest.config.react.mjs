/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",

        // BE
        "src/app/*",
        "src/config/*",
        "src/lib/bff/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.{tsx,ts}"],
      reporter: ["text", "html"],
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
