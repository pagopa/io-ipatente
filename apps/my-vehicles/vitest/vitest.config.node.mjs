import { configDefaults, defineConfig } from "vitest/config";
import tsconfigPaths from "vite-tsconfig-paths";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // FE
        "src/hooks/*",
        "src/layouts/*",
      ],
      extension: configDefaults.coverage.extension
        ?.toString()
        .replace(",.tsx", "")
        .split(","),
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
