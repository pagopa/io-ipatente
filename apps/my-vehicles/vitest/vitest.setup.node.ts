/**
 * MSW configuration for Vitest (used as setupFile in vitest.config.node.ts)
 */
import { loadEnvConfig } from "@next/env";
import { afterAll, afterEach, beforeAll } from "vitest";

import { mswServerTest } from "../mocks/msw-server";

let mswServer: ReturnType<typeof mswServerTest>;
// Start server before all tests
beforeAll(() => {
  loadEnvConfig(process.cwd());
  // this done to correctly load the environment variables on msw handlers
  mswServer = mswServerTest();
  mswServer.listen({ onUnhandledRequest: "warn" });
});

//  Close server after all tests
afterAll(() => mswServer.close());

// Reset handlers after each test `important for test isolation`
afterEach(() => mswServer.resetHandlers());
