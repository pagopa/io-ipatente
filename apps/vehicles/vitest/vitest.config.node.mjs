import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",
        "src/app/api/auth/mocks/**",
        // FE
        "src/common/*",
        "src/hooks/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.ts"],
      reporter: ["text", "html"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    env: loadEnv("test", process.cwd(), ""),
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
