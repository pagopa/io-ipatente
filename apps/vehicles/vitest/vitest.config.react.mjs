/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";
import { loadEnv } from "vite";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // BE
        "src/app/*",
        "src/lib/be/*",
      ],
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/frontend`,
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
