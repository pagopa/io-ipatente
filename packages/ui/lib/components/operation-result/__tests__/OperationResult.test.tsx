import { Cached } from "@mui/icons-material";
import { IllusError } from "@pagopa/mui-italia";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import {
  DATA_TEST_ID_OPERATION_RESULT_PREFIX,
  OperationResult,
} from "../OperationResult";

const mockHandleClick = vi.fn();

const action = {
  endIcon: <Cached />,
  label: "action-label",
  onClick: mockHandleClick,
};
const description = "sample-description";
const illustration = <IllusError />;
const title = "sample-title";

describe("Test OperationResult Component", () => {
  it("Should match the snapshot with all props", () => {
    const comp = render(
      <OperationResult
        action={action}
        description={description}
        illustration={illustration}
        title={title}
      />,
    );
    expect(comp).toMatchSnapshot();
  });

  it("Should call onClick when the action button is clicked", async () => {
    const comp = render(
      <OperationResult
        action={action}
        description={description}
        illustration={illustration}
        title={title}
      />,
    );

    const actionButton = await comp.findByTestId(
      `${DATA_TEST_ID_OPERATION_RESULT_PREFIX}-action`,
    );
    fireEvent.click(actionButton);
    expect(mockHandleClick).toHaveBeenCalledTimes(1);
  });
});
