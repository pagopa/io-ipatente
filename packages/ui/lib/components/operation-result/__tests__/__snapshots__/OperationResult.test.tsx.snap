// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test OperationResult Component > Should match the snapshot with all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-1d569rq-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11e46a8-MuiSvgIcon-root"
            focusable="false"
            role="img"
            viewBox="0 0 120 120"
          >
            <path
              clip-rule="evenodd"
              d="M15.197 31.264a.86.86 0 0 0-.032.099 1.12 1.12 0 0 1-.022.073c-.067.189-.107.387-.119.587 0 .02-.006.039-.012.058-.006.02-.012.04-.012.062v85.714c0 1.184.96 2.143 2.143 2.143h85.714a2.142 2.142 0 0 0 2.143-2.143V2.143A2.143 2.143 0 0 0 102.857 0H45a.22.22 0 0 0-.066.013.232.232 0 0 1-.058.013c-.2.01-.397.05-.585.117-.074.023-.145.05-.215.082a2.068 2.068 0 0 0-.484.325c-.046.034-.091.07-.135.108 0 0-.023.015-.032.025l-27.857 30c-.155.173-.28.369-.371.581ZM42.857 7.6V30h-20.8l20.8-22.4ZM45 34.286H19.286v81.428h81.428V4.286H47.143v27.857c0 1.183-.96 2.143-2.143 2.143Z"
              fill-rule="evenodd"
            />
            <path
              d="M50.694 55.16c.154-.144.286-.309.39-.491.052-.069.1-.14.143-.214a2.034 2.034 0 0 0 0-1.767 1.7 1.7 0 0 0-.144-.214 2.086 2.086 0 0 0-.389-.493c-.028-.025-.039-.062-.069-.086l-10.714-8.571a2.144 2.144 0 0 0-2.679 3.347l5.944 4.758H36.43a2.143 2.143 0 0 0 0 4.285h6.747l-5.944 4.756a2.143 2.143 0 1 0 2.679 3.346l10.714-8.572c.03-.023.04-.06.07-.084ZM83.571 51.429h-6.747l5.944-4.756a2.143 2.143 0 1 0-2.679-3.346l-10.714 8.571c-.03.023-.04.06-.07.085a2.077 2.077 0 0 0-.388.49c-.053.07-.1.141-.144.215a2.034 2.034 0 0 0 0 1.767c.043.074.091.145.144.214.103.182.235.347.389.49.028.026.039.063.069.086l10.714 8.57a2.143 2.143 0 1 0 2.679-3.345l-5.944-4.756h6.747a2.143 2.143 0 0 0 0-4.285ZM60 70.714a19.21 19.21 0 0 0-17.883 12.054 2.144 2.144 0 0 0 3.973 1.607 15.002 15.002 0 0 1 27.82 0 2.141 2.141 0 0 0 3.684.524c.47-.607.582-1.42.289-2.131A19.21 19.21 0 0 0 60 70.714Z"
            />
            <title>
              Errore
            </title>
          </svg>
        </div>
        <div
          class="MuiStack-root css-1cuev0n-MuiStack-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
          >
            sample-title
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            sample-description
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
          data-testid="io-ipatente-operation-result-action"
          tabindex="0"
          type="button"
        >
          action-label
          <span
            class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CachedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m19 8-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-1d569rq-MuiStack-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-11e46a8-MuiSvgIcon-root"
          focusable="false"
          role="img"
          viewBox="0 0 120 120"
        >
          <path
            clip-rule="evenodd"
            d="M15.197 31.264a.86.86 0 0 0-.032.099 1.12 1.12 0 0 1-.022.073c-.067.189-.107.387-.119.587 0 .02-.006.039-.012.058-.006.02-.012.04-.012.062v85.714c0 1.184.96 2.143 2.143 2.143h85.714a2.142 2.142 0 0 0 2.143-2.143V2.143A2.143 2.143 0 0 0 102.857 0H45a.22.22 0 0 0-.066.013.232.232 0 0 1-.058.013c-.2.01-.397.05-.585.117-.074.023-.145.05-.215.082a2.068 2.068 0 0 0-.484.325c-.046.034-.091.07-.135.108 0 0-.023.015-.032.025l-27.857 30c-.155.173-.28.369-.371.581ZM42.857 7.6V30h-20.8l20.8-22.4ZM45 34.286H19.286v81.428h81.428V4.286H47.143v27.857c0 1.183-.96 2.143-2.143 2.143Z"
            fill-rule="evenodd"
          />
          <path
            d="M50.694 55.16c.154-.144.286-.309.39-.491.052-.069.1-.14.143-.214a2.034 2.034 0 0 0 0-1.767 1.7 1.7 0 0 0-.144-.214 2.086 2.086 0 0 0-.389-.493c-.028-.025-.039-.062-.069-.086l-10.714-8.571a2.144 2.144 0 0 0-2.679 3.347l5.944 4.758H36.43a2.143 2.143 0 0 0 0 4.285h6.747l-5.944 4.756a2.143 2.143 0 1 0 2.679 3.346l10.714-8.572c.03-.023.04-.06.07-.084ZM83.571 51.429h-6.747l5.944-4.756a2.143 2.143 0 1 0-2.679-3.346l-10.714 8.571c-.03.023-.04.06-.07.085a2.077 2.077 0 0 0-.388.49c-.053.07-.1.141-.144.215a2.034 2.034 0 0 0 0 1.767c.043.074.091.145.144.214.103.182.235.347.389.49.028.026.039.063.069.086l10.714 8.57a2.143 2.143 0 1 0 2.679-3.345l-5.944-4.756h6.747a2.143 2.143 0 0 0 0-4.285ZM60 70.714a19.21 19.21 0 0 0-17.883 12.054 2.144 2.144 0 0 0 3.973 1.607 15.002 15.002 0 0 1 27.82 0 2.141 2.141 0 0 0 3.684.524c.47-.607.582-1.42.289-2.131A19.21 19.21 0 0 0 60 70.714Z"
          />
          <title>
            Errore
          </title>
        </svg>
      </div>
      <div
        class="MuiStack-root css-1cuev0n-MuiStack-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
        >
          sample-title
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          sample-description
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-11qr2p8-MuiButtonBase-root-MuiButton-root"
        data-testid="io-ipatente-operation-result-action"
        tabindex="0"
        type="button"
      >
        action-label
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CachedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m19 8-4 4h3c0 3.31-2.69 6-6 6-1.01 0-1.97-.25-2.8-.7l-1.46 1.46C8.97 19.54 10.43 20 12 20c4.42 0 8-3.58 8-8h3zM6 12c0-3.31 2.69-6 6-6 1.01 0 1.97.25 2.8.7l1.46-1.46C15.03 4.46 13.57 4 12 4c-4.42 0-8 3.58-8 8H1l4 4 4-4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
